training/forms.py

```diff
@@ -57,63 +33,110 @@ class Meta:
+                'https://training.galaxyproject.org/training-material/'
+                '" target="_blank"> Galaxy Training Network</a>.'),
+            "advertise": (
+                "We will create an event article that will be visible on"
+                f' <a href="{settings.TIAAS_OWNER_SITE}">'
+                f'{settings.TIAAS_OWNER_SITE}</a>'),
+            "gtn_links": (
+                "If you are using official training materials: which ones?"
+                " Please provide the topic and material name, or URLs so we"
+                " can find it."),
+            "non_gtn_links": (
+                "If not using official training materials, can you provide"
+                " URLs to your workflows, or just simply a list of all tool"
+                " IDs that you will run. We need the internal Galaxy tool IDs"
+                " (if you right click a tool in the Galaxy UI and copy link"
+                " location, the link will contain the tool ID)."),
+            "retain_contact": (
+                "If you consent we will retain your information for a longer"
+                f" period of time ({settings.TIAAS_GDPR_RETAIN_EXTRA_MONTHS}"
+                " months). We will use this to contact you regarding"
+                " letters of support for our continued funding, and similar"
+                " matters."),
+            "blogpost": (
+                "Would you be willing to write a blogpost after your event,"
+                " summarising your experience with TIaaS and how it helped"
+                " you?"),
+            "training_identifier": (
+                "A unique name to help identify your training resources."
+                " Please use only lowercase letters, numbers and dash (-)"
+                " characters. 20 characters maximum."),
         }
 
         widgets = {
-            "country": CountrySelectWidget(),
             "start": forms.SelectDateWidget(),
             "end": forms.SelectDateWidget(),
             "title": forms.TextInput(),
-            "description": forms.TextInput(),
+            "description": forms.Textarea(attrs={'rows': 4}),
+            "gtn_links": forms.Textarea(attrs={'rows': 4}),
+            "non_gtn_links": forms.Textarea(attrs={'rows': 4}),
             "attendance": forms.NumberInput(attrs={"min": 1}),
+            "other_requests": forms.Textarea(attrs={'rows': 4}),
         }
+
+    @property
+    def all_error_messages(self):
+        """Return list items from self.errors.
+
+        self.errors is a dict-like collection with values as lists. We only
+        need the list items.
+        """
+        return [e for elist in self.errors.values() for e in elist]
+
+    def clean_start(self):
+        start = self.cleaned_data['start']
+        now = timezone.now().date()
+        validate_start_date(start)
+
+        if (
+                'apology' in self.data
+                and self.data['apology'] == "I am very sorry"):
```

Comment:
This is ugly as, let's set the black linting to a larger width. 88 is archaic, 160 is probably more reasonable.

Reply:
Agreed this is ugly, but I think Galaxy's rules on line length are crazy. I'd be much happier with Black's 88 default if we can make that work. Either way, this can definitely be made prettier.

![image](https://user-images.githubusercontent.com/42562517/192166910-fd25751e-080b-44f7-b7cb-29cb8c7c805d.png)

Reply:
yeah fair, I won't argue against accessibility concerns, feel free to change it to your preference and add to the Makefile.

Reply:
Just ran black and committed - this eyesore is gone.